src/rules/no-large-snapshots.js

```diff
@@ -10,8 +10,29 @@ const reportOnViolation = (context, node) => {
   const startLine = node.loc.start.line;
   const endLine = node.loc.end.line;
   const lineCount = endLine - startLine;
+  const whitelistedSnapshots =
+    context.options &&
+    context.options[0] &&
+    context.options[0].whitelistedSnapshots;
+
+  let isWhitelisted = false;
+  if (whitelistedSnapshots) {
+    const fileName = path.relative(process.cwd(), context.getFilename());
+    const whitelistedSnapshotsInFile = whitelistedSnapshots[fileName];
+    if (whitelistedSnapshotsInFile) {
+      const snapshotName = getStringValue(node.expression.left.property);
+      isWhitelisted =
+        whitelistedSnapshotsInFile.find(name => {
+          if (name.test && typeof name.test === 'function') {
+            return name.test(snapshotName);
+          } else {
+            return name === snapshotName;
+          }
+        }) != null;
+    }
+  }
 
-  if (lineCount > lineLimit) {
+  if (!isWhitelisted && lineCount > lineLimit) {
```

Comment:
won't this ignore all long snapshots in the file, not the specific one from config (`snapshot name 1` in your example)? `isWhitelisted` will be `true` if the name is found in the file, regardless of which snapshot it is

Reply:
no, `isWhiteListed` specific to each test/snapshot, not the entire file.  The tests I wrote should have all these bases covered, and when I run it locally against our codebase it works as expected.  Which is that the long snaps are whitelisted in each .snap file (but not the whole file).  The reason for having the .snap file in the whitelisting, is to avoid name collisions.  Suppose you have 2 different snap files with the same test name.  Keying off the path solves that problem.

Reply:
@SimenB  in other words `isWhitelisted` is only true when the file AND snapshot name match.  not just the filename.  I may be over explaining it, but does that make sense? 😄

Reply:
There's actually a test I wrote that covers this case: https://github.com/benmonro/eslint-plugin-jest/blob/feature/white-list-snapshots/src/rules/__tests__/no-large-snapshots.test.js#L220-L248

the title is about regex, but if you notice, it's got 2 large snapshots in the same file, but only reports on the non-whitelisted snapshot.